buffer/src/main/java/io/netty5/buffer/bytebuffer/NioBuffer.java

```diff
@@ -271,6 +271,42 @@ public void copyInto(int srcPos, Buffer dest, int destPos, int length) {
         InternalBufferUtils.copyToViaReverseLoop(this, srcPos, dest, destPos, length);
     }
 
+    @Override
+    public Buffer writeBytes(byte[] source, int srcPos, int length) {
+        if (source.length < srcPos + length) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+
+        checkWrite(woff, length, true);
+
+        if (hasWritableArray()) {
+            System.arraycopy(source, srcPos, writableArray(), writableArrayOffset(), length);
+        } else {
+            writableBuffer().put(source, srcPos, length);
```

Comment:
`writableBuffer` allocates a slice. We could have a version of `InternalBufferUtils.bbput` that works on arrays, and uses [ByteBuffer.put(int, byte[], int, int)](https://docs.oracle.com/en/java/javase/19/docs/api/java.base/java/nio/ByteBuffer.html#put(int,byte%5B%5D,int,int)) when available.

Reply:
I've added a new bbput variant for this and updated the PR accordingly.